sim/aviron/src/lib/io.zig

```diff
@@ -191,3 +161,32 @@ pub const IO = struct {
         return null;
     }
 };
+
+// Unified byte-addressable device interface used by MemorySpace for RAM and IO
+pub const Device = struct {
+    ctx: *anyopaque,
+    vtable: *const VTable,
+
+    pub const VTable = struct {
+        read8: *const fn (ctx: *anyopaque, idx: usize) u8,
+        write8: *const fn (ctx: *anyopaque, idx: usize, v: u8) void,
+        write_masked: *const fn (ctx: *anyopaque, idx: usize, mask: u8, v: u8) void,
```

Comment:
I wouldn't use `idx: usize` here but `idx: u32` or `idx: u24`.

We're implementing a simulator here and the address size is fixed and not system dependent. The max. value of `usize` depends on the host system, but in this case we must depend on the maximum value available on `AVR`. Which is `u24`.